spec/models/group_assignment_repo_spec.rb

```diff
@@ -35,43 +34,82 @@
-            import_github_repo_url = github_url("/repositories/#{@group_assignment_repo.github_repo_id}/import")
-            expect(WebMock).to have_requested(:put, import_github_repo_url)
-          end
-        end
 
-        describe "#add_team_to_github_repository" do
-          it "adds the team to the repository" do
-            github_repo = GitHubRepository.new(organization.github_client, @group_assignment_repo.github_repo_id)
-            add_github_team_url = github_url("/teams/#{group.github_team_id}/repos/#{github_repo.full_name}")
-            expect(WebMock).to have_requested(:put, add_github_team_url)
+          describe "#create_github_repository" do
+            it "creates the repository on GitHub" do
+              expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
+            end
           end
 
-          context "when students_are_repo_admins is true" do
-            before do
-              group_assignment.update(students_are_repo_admins: true)
-              @group_assignment_repo = GroupAssignmentRepo.create(group_assignment: group_assignment, group: group)
+          describe "#push_starter_code" do
+            it "pushes the starter code to the GitHub repository" do
+              import_github_repo_url = github_url("/repositories/#{@group_assignment_repo.github_repo_id}/import")
+              expect(WebMock).to have_requested(:put, import_github_repo_url)
             end
+          end
 
+          describe "#add_team_to_github_repository" do
             it "adds the team to the repository" do
               github_repo = GitHubRepository.new(organization.github_client, @group_assignment_repo.github_repo_id)
               add_github_team_url = github_url("/teams/#{group.github_team_id}/repos/#{github_repo.full_name}")
-              permission_param = { permission: "admin" }
-              expect(WebMock).to have_requested(:put, add_github_team_url).with(body: hash_including(permission_param))
+              expect(WebMock).to have_requested(:put, add_github_team_url)
+            end
+
+            context "when students_are_repo_admins is true" do
+              before do
+                group_assignment.update(students_are_repo_admins: true)
+                @group_assignment_repo = GroupAssignmentRepo.create(group_assignment: group_assignment, group: group)
+              end
+
+              it "adds the team to the repository" do
+                github_repo = GitHubRepository.new(organization.github_client, @group_assignment_repo.github_repo_id)
+                add_github_team_url = github_url("/teams/#{group.github_team_id}/repos/#{github_repo.full_name}")
+                permission_param = { permission: "admin" }
+                expect(WebMock).to have_requested(:put, add_github_team_url)
+                  .with(body: hash_including(permission_param))
+              end
+            end
+          end
+        end
+
```

Comment:
The tests above this comment didn't change at all. They were just moved up into the context `success`.